src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(String fruit, int quantity) {
+        Storage.fruits.put(fruit, Storage.fruits.get(fruit) - quantity);
```

Comment:
Ensure that the stock does not go negative when subtracting the purchase quantity. Consider adding a check to prevent negative stock values and handle cases where the stock is insufficient.